### PR TITLE
Add tests and comments

### PR DESCRIPTION
**Pull request by mchaisse**

### CODE REVIEW

File: spec/factories/sequence.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Factory
+
+  class << self
+
+    def sequence
+      yield((0..1000).to_a.sample)
```
Comment (epmwgs) [inline spec/factories/sequence.rb]: this has the same effect and don't requires memory.
```ruby
  rand(1000)
```

File: spec/factories/token.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Factory
+
+  class << self
+
+    def token
+      [*'0'..'9', *'a'..'z', *'A'..'Z'].sample(40).join
```
Comment (epmwgs) [inline spec/factories/token.rb]: Is this what you mean?
```ruby
SecureRandom.hex(20)
```

File: spec/support/http_accept_language.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class HttpAcceptLanguage
+  def header
+    "en-US,en;q=0.8,fr;q=0.6"
```
Comment (epmwgs) [inline spec/support/http_accept_language.rb]: I don't see es or es-MX.

File: spec/support/logger_helper.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module LoggerHelper
+  require 'logger'
+
+  def stub_logger(filename = "#{Dir.getwd}/tmp/requests.log")
```
Comment (epmwgs) [inline spec/support/logger_helper.rb]: You can send an stream like STDOUT or even /dev/null, no need of a real file path.

File: spec/tracker_hub/request/notification/hip_chat_spec.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    it { expect(described_class::API_VERSION).to be_a(String) }
+    it { expect(described_class::API_VERSION.frozen?).to be(true) }
+  end
+
+
+  describe 'instantiation' do
+
+    context 'with valid arguments' do
+      it { expect(described_class.new('token', 'room', 'username')).to \
+          be_a(TrackerHub::Request::Notification::HipChat) }
```
Comment (epmwgs) [inline spec/tracker_hub/request/notification/hip_chat_spec.rb]: You better check the `new` method doesn't raise error and the instantiated object not be nil.

File: spec/tracker_hub/request/notification/hip_chat_spec.rb [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      context '#room' do
+        it { expect{subject.room}.to raise_error(NoMethodError) }
+      end
+
+      context '#options' do
+        it { expect{subject.options}.to raise_error(NoMethodError) }
+      end
+    end
+
+    context 'private' do
```
Comment (epmwgs) [inline spec/tracker_hub/request/notification/hip_chat_spec.rb]: Don't text private methods, if they are not there some feature of related should be broken, then we fix it.

File: spec/tracker_hub/request/notification_spec.rb [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        it { expect{subject.timelapse = 84}.to raise_error(NoMethodError) }
+      end
+
+      context '#options' do
+        it { expect(subject.options.keys).to include(:my) }
+        it { expect{subject.options = {}}.to raise_error(NoMethodError) }
+      end
+    end
+
+    context 'private' do
```
Comment (epmwgs) [inline spec/tracker_hub/request/notification_spec.rb]: Don't test private methods.

File: spec/tracker_hub/requests_spec.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      it { expect{ described_class.new({}, 200) }.to raise_error(ArgumentError) }
+    end
+  end
+
+  describe 'attributes' do
+    subject { described_class.new({}, 200, {}) }
+
+    describe 'getter' do
+
+      context 'private' do
```
Comment (epmwgs) [inline spec/tracker_hub/requests_spec.rb]: Don't test private methods.

File: spec/tracker_hub/requests_spec.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      context 'public' do
+        it { expect{ subject.request }.to  raise_error(NoMethodError) }
+        it { expect{ subject.status }.to   raise_error(NoMethodError) }
+        it { expect{ subject.response }.to raise_error(NoMethodError) }
+      end
+    end
+
+    describe 'setter' do
+
+      context 'private' do
```
Comment (epmwgs) [inline spec/tracker_hub/requests_spec.rb]: Don't test private methods

**Comment by mchaisse** [comment]
@epmwgs PTAL

**Pull request merged**